### PR TITLE
Updates to a few Sorcerer skills

### LINKED ISSUE #6387: Diamond Dust skill Can Overlap
* **rAthena Hash**: https://github.com/rathena/rathena/commit/3d4021bfc61be0319aed26e5b496b80faa0e5f88

* **Client Date**: 2020-09-02

* **Server Mode**: RE

* **Description of Issue**: 
  * Result: 
The Diamond Dust skill has a NoOverlap mapflag.

  
  * Expected Result: 
The Diamond Dust skill can overlap.

  
  * How to Reproduce: 

  * Official Information: 
  
  
kRO video
https://youtu.be/bwIqfOeecw4?t=54

* **Modifications that may affect results**:

### LINKED ISSUE #6425: Diamond Dust and others Sorcerer's skills has wrong formulas
* **rAthena Hash**: 

153a83d

* **Client Date**: 

2021-11-17

* **Server Mode**: 

Renewal

* **Description of Issue**: 
I compared damage of Sorcerer's skills inside rAthena and at Sakray, damage was different with the same stats
(1st screenshot - rAthena, 2nd - Sakray) 
image
image

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #6425 and fixes #6387

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Updates the damage formulas for Earth Grave, Diamond Dust, and Varetyr Spear.
  * Removes the NoOverlap flag for Diamond Dust and Comet.
Thanks to @Rayvakarian, @Everade, @Balferian, and @kuranwinterfall!

### CODE REVIEW

**Comment by Rayvakarian** [comment]
Can you add overlap for Diamond Dust ?https://github.com/rathena/rathena/issues/6387

File: src/map/battle.cpp [hunk 1]
```diff
@@ -7177,14 +7177,14 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
 					case NPC_ELECTRICWALK:
 						skillratio += -100 + 100 * skill_lv;
 						break;
-					case SO_EARTHGRAVE: // !TODO: Confirm formula
-						skillratio += -100 + sstatus->int_ / 6 * skill_lv + ((sd) ? pc_checkskill(sd, SA_SEISMICWEAPON) * 200 : 0);
+					case SO_EARTHGRAVE:
+						skillratio += -100 + 2 * sstatus->int_ + 300 * ((sd) ? pc_checkskill(sd, SA_SEISMICWEAPON) : 0) + sstatus->int_ * skill_lv;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:7181]: The short if for sd could be removed. The first check inside pc_checkskill is if sd is nullptr and return 0 in this case.

File: src/map/battle.cpp [hunk 2]
```diff
@@ -7177,14 +7177,14 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
...
 						RE_LVL_DMOD(100);
 						if( sc && sc->data[SC_CURSED_SOIL_OPTION] )
 							skillratio += (sd ? sd->status.job_level * 5 : 0);
 						break;
-					case SO_DIAMONDDUST: // !TODO: Confirm formula
-						skillratio += -100 + 200 * ((sd) ? pc_checkskill(sd, SA_FROSTWEAPON) : 0) + sstatus->int_ / 6 * skill_lv;
+					case SO_DIAMONDDUST:
+						skillratio += -100 + 2 * sstatus->int_ + 300 * ((sd) ? pc_checkskill(sd, SA_FROSTWEAPON) : 0) + sstatus->int_ * skill_lv;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:7187]: The short if for sd could be removed. The first check inside pc_checkskill is if sd is nullptr and return 0 in this case.

File: src/map/battle.cpp [hunk 3]
```diff
@@ -7224,8 +7224,8 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
 					case NPC_CLOUD_KILL:
 						skillratio += -100 + 50 * skill_lv;
 						break;
-					case SO_VARETYR_SPEAR: //MATK [{( Endow Tornado skill level x 50 ) + ( Caster INT x Varetyr Spear Skill level )} x Caster Base Level / 100 ] %
-						skillratio += -100 + sstatus->int_ / 6 * skill_lv + ((sd) ? pc_checkskill(sd, SA_LIGHTNINGLOADER) * 50 : 0); // !TODO: Confirm new formula
+					case SO_VARETYR_SPEAR:
+						skillratio += -100 + (2 * sstatus->int_ + 150 * ((sd ? pc_checkskill(sd, SO_STRIKING) : 0) + (sd ? pc_checkskill(sd, SA_LIGHTNINGLOADER) : 0)) + sstatus->int_ * skill_lv / 2) / 3;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:7228]: The short if for sd could be removed. The first check inside pc_checkskill is if sd is nullptr and return 0 in this case.

**Review by Lemongrass3110 — APPROVED** [review]
Looks good. Just some thoughts for readability.

**Pull request merged**